crates/polars-python/src/functions/lazy.rs

```diff
@@ -442,12 +442,24 @@ pub fn lit(value: &Bound<'_, PyAny>, allow_object: bool) -> PyResult<PyExpr> {
         });
         Ok(dsl::lit(s).into())
     } else {
-        Err(PyTypeError::new_err(format!(
-            "cannot create expression literal for value of type {}: {}\
-            \n\nHint: Pass `allow_object=True` to accept any value and create a literal of type Object.",
-            value.get_type().qualname()?,
-            value.repr()?
-        )))
+        Python::with_gil(|py| {
+            // One final attempt before erroring. Do we have a date/datetime subclass?
+            // E.g. pd.Timestamp, or Freezegun.
+            let datetime_module = PyModule::import_bound(py, "datetime")?;
+            let datetime_class = datetime_module.getattr("datetime")?;
+            let date_class = datetime_module.getattr("date")?;
+            if value.is_instance(&datetime_class)? || value.is_instance(&date_class)? {
+                let av = py_object_to_any_value(value, true)?;
+                Ok(Expr::Literal(LiteralValue::try_from(av).unwrap()).into())
```

Comment:
Pyo3 does have `PyDateTime`, but it's limited to:

> Available on non-Py_LIMITED_API only

Reply:
This should come before the `allow_object` check. 

I am thinking aloud here. Why don't we go via `AnyValue` directly for all branches? This seems like code duplication.

Reply:
yup, thanks for catching that, and sorry for not having spotted it myself

fixed, and I've added a test which would've on my previous commit to make sure this doesn't regress

Reply:
> Why don't we go via AnyValue directly for all branches?

Tried then, but then a float would become
```
[crates/polars-python/src/functions/lazy.rs:429:9] &newret = Ok(
    PyExpr {
        inner: 2.0,
    },
)
```
instead of
```
[crates/polars-python/src/functions/lazy.rs:432:9] &oldret = Ok(
    PyExpr {
        inner: dyn float: 2.0,
    },
)
```
, and I haven't (yet) dived deeper here

Reply:
Ah, yeap.. That's a good reason.